tests/test_inference_providers.py

```diff
@@ -1252,6 +1344,14 @@ def test_recursive_merge(dict1: Dict, dict2: Dict, expected: Dict):
             {"a": [0, 1, None]},  # do not remove None in lists
             {"a": [0, 1, None]},
         ),
+        # dicts inside list are cleaned, list level None kept
+        ({"a": [{"x": None, "y": 1}, None]}, {"a": [{"y": 1}, None]}),
+        # remove every None that is the value of a dict key
+        (
+            [None, {"x": None, "y": 5}, [None, 6]],
+            [None, {"y": 5}, [None, 6]],
+        ),
+        ({"a": [None, {"x": None}]}, {"a": [None]}),
```

Comment:
```suggestion
        ({"a": [None, {"x": None}]}, {"a": [None], {}}),
```

Very much a corner case here but I think it's preferable to keep the same list length in case of empty list. It feels less error-prone since we are always keeping None values in lists.

Reply:
makes sense! fixed in [03019ff](https://github.com/huggingface/huggingface_hub/pull/3178/commits/03019ff38730d64e92e8e7447c5f4f5648cb6d2f)